python311Packages.apple-weatherkit: init at 1.0.3

fabaff opened the pull request:
## Description of changes
Python library for Apple WeatherKit

https://github.com/tjhorner/python-weatherkit
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

fabaff committed:
python311Packages.apple-weatherkit: init at 1.0.3

Python library for Apple WeatherKit
https://github.com/tjhorner/python-weatherkit

fabaff commented:
Result of `nixpkgs-review pr 255541` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python310Packages.apple-weatherkit</li>
    <li>python310Packages.apple-weatherkit.dist</li>
    <li>python311Packages.apple-weatherkit</li>
    <li>python311Packages.apple-weatherkit.dist</li>
  </ul>
</details>

lilyball commented:
Note that the upstream release is now 1.0.4 (and HomeAssistant 2023.10 wants version 1.0.4)

lilyball commented:
Also this works for me if I add `pyjwt.optional-dependencies.crypto` to the deps list (and bump the version to 1.0.4).

fabaff committed:
python311Packages.apple-weatherkit: 1.0.3 -> 1.0.4

Diff: https://github.com/tjhorner/python-weatherkit/compare/refs/tags/v1.0.3...v1.0.4

Changelog: https://github.com/tjhorner/python-weatherkit/releases/tag/v1.0.4

fabaff commented:
Result of `nixpkgs-review pr 255541` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python311Packages.apple-weatherkit</li>
    <li>python311Packages.apple-weatherkit.dist</li>
  </ul>
</details>

fabaff committed:
home-assistant: update component-packages

fabaff commented:
Result of `nixpkgs-review pr 255541` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>home-assistant-component-tests.weatherkit</li>
    <li>python311Packages.apple-weatherkit</li>
    <li>python311Packages.apple-weatherkit.dist</li>
  </ul>
</details>
